### DISCUSSION TITLE
Is Apollo Server v4 supported?

### DISCUSSION THREAD

**Discussion by jbeckton** [MichalLytek/typegraphql-prisma#441] [General]
https://prisma.typegraphql.com/docs/basics/usage#context-configuration

Following this doc on your site, it says you can use v4 

> Below you can find an example using Apollo Server V4:

Code snippet is clearly not gonna work with v4

````
import { PrismaClient } from "@prisma/client";
import { ApolloServer } from "@apollo/server";
import { startStandaloneServer } from "@apollo/server/standalone";

const prisma = new PrismaClient();

const server = new ApolloServer<MyContext>({
  schema, // from previous step
});
const { url } = await startStandaloneServer(server, {
  context: () => ({ prisma }),
  listen: { port: 4000 },
});
````

v4 requires typeDefs,

`const server = new ApolloServer<MyContext>({ typeDefs, resolvers });`

Can't find any docs regarding where to import the typeDefs from, looking around in generated code but I see none.

**Comment by jbeckton** [comment]
And then there is this...

Type 'GraphQLSchema' is missing the following properties from type 'GraphQLSchema': __validationErrors, _queryType, _mutationType, _subscriptionType, and 6 more.ts(2740)
constructor.d.ts(53, 5): The expected type comes from property 'schema' which is declared here on type 'ApolloServerOptions<Context>'

```` 
// Create GraphQL server
  const server = new ApolloServer<Context>({
    schema
  });
````

**Comment by MichalLytek** [comment]
> v4 requires typeDefs,

That's not true.

https://github.com/MichalLytek/type-graphql/blob/master/examples/simple-usage/index.ts

**Reply by jbeckton** [reply]
ok, my mistake... can you help me figure out what I missed? I followed the guide referenced in the link in my first post and I now have this issue.

![image](https://github.com/MichalLytek/typegraphql-prisma/assets/2669585/a3a3e387-161b-4df2-a9d7-63062c374074)


```` 
Type 'GraphQLSchema' is missing the following properties from type 'GraphQLSchema': __validationErrors, _queryType, _mutationType, _subscriptionType, and 6 more.ts(2740)
constructor.d.ts(53, 5): The expected type comes from property 'schema' which is declared here on type 'ApolloServerOptions<Context>'
````

**Reply by jbeckton** [reply]
disregard, I think this issue is caused by something being mis-configured in my mono-repo project. Creating a new project outside of the mono-repo seems to works fine.  I'll have to do further investigation on my end.

Thanks

**Reply by MichalLytek** [reply]
Yes, it's because your monorepo has different versions of `graphql-js` dependency
